app/views/mailing_list/steps/_form.html.erb

```diff
@@ -7,9 +7,9 @@
     <%= render current_step.key, current_step: current_step, f: f %>
 
     <% if wizard.can_proceed? %>
-      <%= f.button class: "call-to-action-button button" do %>
-        <%= wizard.last_step? ? "Complete sign up" : "Next Step" %>
-        <span></span>
+      <% button_text = safe_html_format(wizard.last_step? ? "Complete sign up <span></span>" : "Next Step <span></span>") %>
+      <%= f.button class: "call-to-action-button button", data: { "prevent-double-click": true, "disable-with": button_text } do %>
+        <%= button_text %>
       <% end %>
```

Comment:
This could be shortened a bit

```suggestion
      <%= f.button button_text, class: "call-to-action-button button", data: { "prevent-double-click": true, "disable-with": button_text } %>
```

Reply:
👍 updated